datastore/sync_entity.go

```diff
@@ -594,15 +599,16 @@ func (dynamo *Dynamo) UpdateSyncEntity(entity *SyncEntity, oldVersion int64) (bo
 		ExpressionAttributeValues: expr.Values(),
 		ConditionExpression:       expr.Condition(),
 		UpdateExpression:          expr.Update(),
-		ReturnValues:              aws.String(dynamodb.ReturnValueAllOld),
+		ReturnValues:              types.ReturnValueAllOld,
 		TableName:                 aws.String(Table),
 	}
 
-	out, err := dynamo.UpdateItem(input)
+	out, err := dynamo.UpdateItem(context.TODO(), input)
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) {
 			// Return conflict if the write condition fails.
-			if aerr.Code() == dynamodb.ErrCodeConditionalCheckFailedException {
+			if apiErr.ErrorCode() == "ConditionalCheckFailedException" {
```

Comment:
Could also be checked against the correct type like suggested above. I noticed you're already using the error types in other places, so it would be good to stay consistent.